src/device_state/macos/mod.rs

```diff
@@ -116,16 +121,36 @@ impl DeviceState {
             readmouse::Mouse::Center.is_pressed(),
             false,
         ];
+
         MouseState {
             coords: (x as i32, y as i32),
             button_pressed,
         }
     }
+
     pub fn query_keymap(&self) -> Vec<Keycode> {
         MAPPING
             .iter()
             .filter(|(from, _)| from.is_pressed())
-            .map(|(_, to)| to.clone())
+            .map(|(_, to)| *to)
             .collect()
     }
 }
+
+/// Returns true if the Accessibility permissions necessary for this library to work are granted
+/// to this process
+///
+/// If this returns false, the app can request them through the OS APIs, or the user can:
+///   1. open the MacOS system preferences
+///   2. go to Security -> Privacy
+///   3. scroll down to Accessibility and unlock it
+///   4. Add the app that is using device_query (such as your terminal) to the list
+///
+fn has_accessibility() -> bool {
+    use self::macos_accessibility_client::accessibility::*;
+    // Without prompting:
+    // application_is_trusted()
+
+    // With prompting:
+    application_is_trusted_with_prompt()
+}
```

Comment:
This is the meat of the change. Everything else is just touching up nits or making this work.